R/ImportMethods.R

```diff
@@ -500,6 +503,82 @@ import.CTSS <- function(filepath) {
   gp
 }
 
+
+#' Read in BigWig files to CAGEexp object
+#'
+#' @param genome the name of the reference genome (bsgenome)
+#' @param filepath list of input bigwig files with full path
+#' The filenames should include a "_str1" and "_str2" substring after the sample name
+#' The sampe name will be the prefix before these substrings
+#' the filepath in the CAGEexp object will only include the bw corresponding to str1
+#' 
+#' @return a CAGEexp object
+#' 
+#' @family loadFileIntoGPos
+#' 
+#' @author Katalin Ferenc
+#' @author Damir Baranasic
+#' 
+#' @examples
+#' import.bigwig(
+#'  genome="BSgenome.Drerio.UCSC.danRer7",
+#'  filepath=system.file("extdata", "NCig10061_subsampled_str1.Signal.Unique.str1.out.wig.bw")
+#'  )
+
+import.bigwig <- function(
+    genome,
+    filepath){
+
+    str2_path <- gsub("str1","str2" , filepath)
+    str1_str2_paths <- c(filepath, str2_path)
+    signals <- lapply(
+        str1_str2_paths,
+        function(x) {
+            track_in <- rtracklayer::import(x)
+            track_in
+        })
+
+    signal_names <- sub("(_str1|_str2).*", "\\1", basename(str1_str2_paths))
+    names(signals) <- signal_names
+
+    # divide the signals into plus and minus strands
+    signalsSplit <- split(
+        signals,
+        grepl("str1", names(signals)))
+    plus <- lapply(signalsSplit$`TRUE`, function(x) {
+        strand(x) = "+"
+        return(x)
+    })
+    minus <- lapply(signalsSplit$`FALSE`, function(x) {
+        strand(x) = "-"
+        return(x)
+    })
+
+    plus_sample_names <- gsub("_str1", "", names(plus))
+    minus_sample_names <- gsub("_str2", "", names(minus))
+
+    names(plus) <- plus_sample_names
+    names(minus) <- minus_sample_names
+
+    if (!all(plus_sample_names == minus_sample_names)) {
+        if (setequal(plus_sample_names, minus_sample_names)) {
+            minus = minus[match(plus_sample_names, minus_sample_names)]
+        } else {
+            stop("Error: Some basenames of minus- and plus-strand bigWigs are different! Are these bigWigs from different sets of samples? Exit.")
+        }
+    }
+
+    # Load each file as GRangesList where each GRange is a CTSS data.
+    merged <- mapply(c, plus, minus)
+    merged_gpos <- lapply(merged, function(x) {
+        gp <- GPos(stitch=FALSE, x)
+        score(gp) <- x$score
+    })
+
+    merged_gpos
```

Comment:
A lot of your code here has un-necessary complications because your original function supported loading from more than one pair of files, while here it only loads from one pair, and `getCTSS` takes care of looping and naming.  The following stub would do most of the work:

```
    str2_path <- gsub("str1","str2" , filepath)
    if(!file.exists(str2_path)) stop("File ", str2_path, " does not exist!")
    plus <- import.bw(filepath)  # Please import the function in the CAGEr's namespace using Roxygen
    strand(plus) <- '+'
    minus <- import.bw(str2_path) 
    strand(minus) <- '-'
    GPos(c(plus, minus), stitch=FALSE) # I do not remember if the import functions are expected to sort their output…
```